src/components/schema-editor/schema-editor.component.tsx

```diff
@@ -84,21 +111,134 @@ const SchemaEditor: React.FC<SchemaEditorProps> = ({ invalidJsonErrorMessage, on
+        const pathSegments = schemaPath.split('/').filter((segment) => segment !== '' || segment !== 'type');
+
+        let lineNumber = -1;
+
+        for (const segment of pathSegments) {
+          if (segment === 'properties' || segment === 'items') continue; // Skip 'properties' and 'items'
+          const match = segment.match(/^([^[\]]+)/); // Extract property key
+          if (match) {
+            const propertyName: string = pathSegments[pathSegments.length - 2]; // Get property key
+            lineNumber = jsonLines.findIndex((line) => line.includes(propertyName));
+          }
+          if (lineNumber !== -1) break;
+        }
+
+        return lineNumber;
+      };
+
+      if (!isValid) {
+        const errorMarkers = validate.errors.map((error) => {
+          const schemaPath = error.schemaPath.replace(/^#\//, ''); // Remove leading '#/'
+          const lineNumber = traverse(schemaPath);
+          const pathSegments = error.instancePath.split('.'); // Split the path into segments
+          const errorPropertyName = pathSegments[pathSegments.length - 1];
+
+          const message =
+            error.keyword === 'type'
+              ? `${errorPropertyName.charAt(0).toUpperCase() + errorPropertyName.slice(1)} ${error.message}`
+              : `${error.message.charAt(0).toUpperCase() + error.message.slice(1)}`;
+
+          return {
+            startRow: lineNumber,
+            startCol: 0,
+            endRow: lineNumber,
+            endCol: 1,
+            className: 'error',
+            text: message,
+            type: 'error',
+          };
+        });
+
+        setErrors(errorMarkers);
+      } else {
+        setErrors([]);
+      }
+    } catch (error) {
+      console.log('Error during JSON parsing or validation:', error); // eslint-disable-line
+    }
+  };
+
+  const debouncedValidateJSON = debounce(validateJSON, 300);
+
+  const handleChange = (newValue: string) => {
+    setValidation(false);
+    setCurrentIndex(0);
+    onSchemaChange(newValue);
+    debouncedValidateJSON(newValue, schema);
+  };
+
+  // Schema Validation Errors
+  const ErrorNotification = ({ text, line }) => (
+    <ActionableNotification
+      subtitle={text}
+      inline={true}
+      title={`Error, line ${line + 1}:`}
+      kind="error"
+      lowContrast={true}
+      onActionButtonClick={() => window.open('https://json.openmrs.org/form.schema.json', '_blank')}
+      actionButtonLabel={t('referenceSchema', 'Reference schema')}
+    />
+  );
+
+  const onPreviousErrorClick = () => {
+    setCurrentIndex((prevIndex) => Math.max(prevIndex - 1, 0));
+  };
+
+  const onNextErrorClick = () => {
+    setCurrentIndex((prevIndex) => Math.min(prevIndex + 1, errors.length - 1));
+  };
+
+  const ErrorMessages = () =>
+    errors.length > 0 && (
+      <div className={styles.validationErrorsContainer}>
+        <ErrorNotification text={errors[currentIndex].text} line={errors[currentIndex].startRow} />
```

Comment:
```suggestion
        <ErrorNotification text={errors[currentIndex]?.text} line={errors[currentIndex]?.startRow} />
```

Did we see on the call today that we might need to use optional chaining to access these values?